### PR TITLE
Rename NonSimplifiable to ConstructorLike

### PR DESCRIPTION
**Pull request by ttuegel**
I hope this will end a regrettable source of confusion.

I also removed three unused modules.

Fixes: #1309 

---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Attribute/Pattern/ConstructorLike.hs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    synthetic = const (ConstructorLike Nothing)
+    {-# INLINE synthetic #-}
+
+instance Synthetic ConstructorLike (Ceil sort) where
+    synthetic = const (ConstructorLike Nothing)
+    {-# INLINE synthetic #-}
+
+-- A domain value is not technically a constructor, but it is
+-- constructor-like for builtin domains, at least from the
+-- perspective of normalization (normalized means non-simplifiable here).
```
Comment (ana-pantilie) [inline kore/src/Kore/Attribute/Pattern/ConstructorLike.hs]:
```suggestion
-- perspective of normalization (normalized means constructor-like here).
```

File: kore/src/Kore/Attribute/Pattern/ConstructorLike.hs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    synthetic = const (ConstructorLike Nothing)
+    {-# INLINE synthetic #-}
+
+instance Synthetic ConstructorLike (Rewrites sort) where
+    synthetic = const (ConstructorLike Nothing)
+    {-# INLINE synthetic #-}
+
+-- A domain value is not technically a constructor, but it is
+-- constructor-like for builtin domains, at least from the
+-- perspective of normalization (normalized means non-simplifiable here).
```
Comment (ana-pantilie) [inline kore/src/Kore/Attribute/Pattern/ConstructorLike.hs]:
```suggestion
-- perspective of normalization (normalized means constructor-like here).
```

File: kore/src/Kore/Attribute/Symbol.hs [hunk 3]
```diff
@@ -182,8 +182,8 @@ instance Default Symbol where
     def = defaultSymbolAttributes
 
 -- | Is a symbol non-simplifiable?
-isNonSimplifiable :: StepperAttributes -> Bool
-isNonSimplifiable = do
+isConstructorLike :: StepperAttributes -> Bool
```
Comment (ana-pantilie) [inline kore/src/Kore/Attribute/Symbol.hs]: The comments above and below should rename `non-simplifiable` to `constructor-like`.

**Comment by ana-pantilie** [comment]
This looks great, my suggestions are only in a few places in the comments where we still use non-simplifiable. Also, I think you forgot to add the definition of `constructor-like` to `glossary.md`.

File: kore/src/Kore/Attribute/Pattern/ConstructorLike.hs [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    )
+
+{- | A pattern is 'ConstructorLike' if logical equality is syntactic equality.
+
+In other words, a pattern is constructor-like if it is equal (in the logical
+'Equals' sense) to another constructor-like pattern if and only if it is
+syntactically equal (in the 'Eq' sense).
+
+Examples of patterns that are constructor-like:
+- 'BuiltinBool', 'BuiltinInt', 'BuiltinString', and 'InternalBytes'
```
Comment (virgil-serbanuta) [inline kore/src/Kore/Attribute/Pattern/ConstructorLike.hs]: Should a BuiltinMap with only constructor-like things in it be constructor-like?

**Pull request merged**